src/public/app/widgets/type_widgets/canvas_note.js

```diff
@@ -0,0 +1,514 @@
+                width: excalidrawWrapperRef.current.getBoundingClientRect().width,
+                height: excalidrawWrapperRef.current.getBoundingClientRect().height
+            };
+            this.log('effect, setdimensions', dimensions);
+            setDimensions(dimensions);
+
+            const onResize = () => {
+                const dimensions = {
+                    width: excalidrawWrapperRef.current.getBoundingClientRect().width,
+                    height: excalidrawWrapperRef.current.getBoundingClientRect().height
+                };
+                this.log('onResize, setdimensions', dimensions);
+                setDimensions(dimensions);
+            };
+            
+            window.addEventListener("resize", onResize);
+            // ensure that resize is also called for split creation and deletion
+            // not really the problem. problem is saved appState!
+            // self.$renderElement.addEventListener("resize", onResize);
+            
+            return () => window.removeEventListener("resize", onResize);
+        }, [excalidrawWrapperRef]);
+
+        const onLinkOpen = React.useCallback((element, event) => {
+            const link = element.link;
+            const { nativeEvent } = event.detail;
+            const isNewTab = nativeEvent.ctrlKey || nativeEvent.metaKey;
+            const isNewWindow = nativeEvent.shiftKey;
+            const isInternalLink =
+                link.startsWith("/") || link.includes(window.location.origin);
+                
+            this.log("onLinkOpen", element, event, nativeEvent, "isinternallink", isInternalLink);
+
+            if (isInternalLink && !isNewTab && !isNewWindow) {
+                // signal that we're handling the redirect ourselves
+                event.preventDefault();
+                // do a custom redirect, such as passing to react-router
+                // ...
+            } else {
+                // open in same tab
+            }
+          }, []);
+
+        return React.createElement(
+            React.Fragment,
+            null,
+            React.createElement(
+                "div",
+                {
+                    className: "excalidraw-wrapper",
+                    ref: excalidrawWrapperRef
+                },
+                React.createElement(Excalidraw.default, {
+                    ref: excalidrawRef,
+                    width: dimensions.width,
+                    height: dimensions.height,
+                    // initialData: InitialData,
+                    onPaste: (data, event) => {
+                        this.log("excalidraw internal paste", data, event);
+                    },
+                    onChange: debounce(this.onChangeHandler, this.debounceTimeOnchangeHandler),
+                    // onPointerUpdate: (payload) => console.log(payload),
+                    onCollabButtonClick: () => {
+                        window.alert("You clicked on collab button")
+                    },
```

Comment:
Is it possible to outright remove the collab button? It might be more helpful to say something like "Collaboration disabled within Trilium" if not, rather than just "you clicked it".

Reply:
I am not sure. I think it would be good to disable some buttons and stuff. It is not possible via configuration, I think. We could hide them via CSS maybe, which is quite hacky.

Now, when I think about it, the Collab button could be used to enable shareing? Or maybe we can open a box that shows all the links to embed the canvas-note (api/image..., share/api/image...)

Reply:
The collab button can be used to actually collaborate; it is a pretty small [server](https://github.com/excalidraw/excalidraw-room). I'm not sure if it's in scope though (probably not) but it would be pretty cool.

Regardless the error message could be changed to something a bit more descriptive in case someone sees that and thinks it is enabled

Reply:
I think in the near term it would be nice to hide such buttons, with CSS if needed. But it's not a blocker.